src/lib/web.sh

```diff
@@ -49,3 +49,26 @@ function replace_http_by_https()
 
   return "$ret"
 }
+
+# This function is a predicate to determine if a file is a HTML file. The function
+# tries to do this efficiently by checking only the first line of the file.
+#
+# @file_path: Path to the file to be checked.
+#
+# Return:
+# Returns 0 if the function decided that the file is a HTML file, 1 if the function
+# decided it isn't, and 2 (ENOENT) if `@file_path` doesn't correspond to a file.
+function is_html_file()
+{
+  local file_path="$1"
+  local first_line_of_file
+
+  [[ ! -f "$file_path" ]] && return 2 # ENOENT
+
+  first_line_of_file=$(head -n 1 "$file_path" | tr '[:upper:]' '[:lower:]')
+  if [[ "$first_line_of_file" =~ ^(<html|<\!doctype html>) ]]; then
```

Comment:
This is usually true; I guess this will always work for the lore use case 100%. However, in terms of URLs, this is not always true. I would add more checks if the first check fails to make this function more robust. For example, you can look for `<header>` and `<body>`; if this function does not find anything like that, then you return false.